src/helpers.js

```diff
@@ -0,0 +1,27 @@
+export function getCharCodes(string) {
+	const codes = new Uint8Array(string.length);
+
+	for (let i = 0, length = string.length; i < length; i++) {
+		codes[i] = string.charCodeAt(i) & 0xff;
+	}
+	return codes;
+}
+
+// mostly internal function for wrapping any supported input (String or Array-like) to best suitable buffer format
+export function wrapBuffer(buffer) {
+	switch (typeof buffer) {
+		case "number":
+			buffer = new Uint8Array(buffer).buffer;
+
+			break;
+
+		case "string":
+			buffer = getCharCodes(buffer);
+		/* falls through */
```

Comment:
If `getCharCodes` now creates its own `Uint8Array`, there's no point in falling through to those extra checks. Let's just do `buffer = getCharCodes(buffer).buffer;` here and `break`.